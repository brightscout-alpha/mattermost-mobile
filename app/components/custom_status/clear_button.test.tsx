// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {shallow} from 'enzyme';
import React from 'react';
import Preferences from '@mm-redux/constants/preferences';
import CompassIcon from '@components/compass_icon';
import ClearButton from '@components/custom_status/clear_button';

describe('components/custom_status/custom_status_emoji', () => {
    const baseProps = {
        theme: Preferences.THEMES.default,
        handlePress: jest.fn(),
    };

    it('should match snapshot', () => {
        const wrapper = shallow(
            <ClearButton
                {...baseProps}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should call handlePress when press event is fired', () => {
        const wrapper = shallow(
            <ClearButton
                {...baseProps}
            />,
        );

        wrapper.find(CompassIcon).simulate('press');
        expect(baseProps.handlePress).toBeCalled();
    });
});
