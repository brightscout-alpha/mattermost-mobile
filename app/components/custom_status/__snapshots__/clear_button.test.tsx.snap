// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/custom_status/custom_status_emoji should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 40,
      "justifyContent": "center",
      "width": 40,
    }
  }
>
  <CompassIcon
    color="#ffffff"
    name="close"
    onPress={[Function]}
    size={20}
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.52)",
        "borderRadius": 1000,
        "overflow": "hidden",
      }
    }
  />
</View>
`;
